### PR TITLE
:sparkles: Gradle support

### PR DESCRIPTION
**Pull request by jmle**
(https://github.com/konveyor/analyzer-lsp/issues/598) Adds support for fetching dependencies with Gradle projects.

- This implementation assumes that there is a Gradle wrapper included in the project, otherwise it is not possible to know which Gradle version is needed.
- To maximize compatibility with most Gradle versions, JDK8 must be used. There are a number of ways to specify this on newer Gradle versions, but for older versions the only thing that seems to work is setting `JAVA_HOME`.
- Depends on https://github.com/konveyor/java-analyzer-bundle/pull/93

:exclamation: This PR does not provide support for fetching dependency sources yet.

:green_circle: Tested against https://github.com/andyjduncan/gradle-example and https://github.com/kissaten/gradle-multi-project-example

### CODE REVIEW

File: demo-dep-output.yaml [hunk 1]
```diff
@@ -2332,4 +2463,4 @@
     labels:
     - konveyor.io/dep-source=open-source
     - konveyor.io/language=java
-    prefix: file:///root/.m2/repository/org/yaml/snakeyaml/1.30
+    prefix: file://org/yaml/snakeyaml/1.30
```
Comment (pranavgaikwad) [inline demo-dep-output.yaml:2466]: what happened to file URIs here?

File: external-providers/java-external-provider/pkg/java_external_provider/dependency.go [hunk 2]
```diff
@@ -51,10 +67,41 @@ func (p *javaServiceClient) findPom() string {
...
+	if _, err := os.Stat(f); errors.Is(err, os.ErrNotExist) {
+		return ""
+	}
 	return f
 }
 
+func (p *javaServiceClient) findGradleBuild() string {
+	// TODO: naive?
+	if p.config.Location != "" {
+		f, err := filepath.Abs(filepath.Join(p.config.Location, "build.gradle"))
```
Comment (pranavgaikwad) [inline external-providers/java-external-provider/pkg/java_external_provider/dependency.go:79]: is Abs() running os.Stat()?

File: external-providers/java-external-provider/pkg/java_external_provider/dependency.go [hunk 3]
```diff
@@ -51,10 +67,41 @@ func (p *javaServiceClient) findPom() string {
...
+		if err != nil {
+			return ""
+		}
+		return f
+	}
+	return ""
+}
+
 func (p *javaServiceClient) GetDependencies(ctx context.Context) (map[uri.URI][]*provider.Dep, error) {
+	if p.getBuildTool() == gradle {
```
Comment (pranavgaikwad) [inline external-providers/java-external-provider/pkg/java_external_provider/dependency.go:89]: is this redundant given that there's a switch in GetDepenenciesDAG()? I think we will need to move fallback call inside GetDepenciesDAG()

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
small nits / questions...overall looks great!

File: external-providers/java-external-provider/pkg/java_external_provider/dependency.go (hunk 2 above)
Comment (jmle) [reply external-providers/java-external-provider/pkg/java_external_provider/dependency.go:79]: :+1:

File: demo-dep-output.yaml (hunk 1 above)
Comment (jmle) [reply demo-dep-output.yaml]: This should be fixed now

**Comment by jmle** [comment]
Superseded by https://github.com/konveyor/analyzer-lsp/pull/609

**Pull request closed without merging**